### PR TITLE
solve `modelsv2.DryRunRequest` type issue from algob

### PR DESCRIPTION
**Pull request by ratik21**
Still had to do a type hack :/

### CODE REVIEW

File: packages/algob/src/lib/dryrun.ts [hunk 1]
```diff
@@ -266,6 +176,18 @@ export class Tealdbg {
...
+
+    /*
+      Encoding fails on taking empty arrays ([]), so we need to convert
+      to undefined first.
+    */
+    if (dryRunRequest.accounts.length === 0) {
+      (dryRunRequest as any).accounts = undefined;
+    }
+    if (dryRunRequest.apps.length === 0) {
+      (dryRunRequest as any).apps = undefined;
```
Comment (robert-zaremba) [inline packages/algob/src/lib/dryrun.ts:188]: can you try to not use `as`? We want to limit it, as it already created few problems.

**Review by robert-zaremba — COMMENTED** [review]
Left one comment about the `as` usage.

**Comment by robert-zaremba** [comment]
test is also not passing

File: packages/algob/src/lib/dryrun.ts (hunk 1 above)
Comment (ratik21) [reply packages/algob/src/lib/dryrun.ts:188]: [type hack] Need to do it here, as `dryRunReq.apps` has type `modelsv2.Application[]` (but it should be `Application[] | undefined`) here. If you don't set it as `undefined`, `encodeObj` will fail.

I can create an issue on their repo though

**Comment by ratik21** [comment]
> test is also not passing

yeah that's a separate issue with workflow. Tests are failing in all branches
image

**Comment by robert-zaremba** [comment]
I think i fixed tests in develop. Need to recheck tomorrow

File: packages/algob/src/lib/dryrun.ts (hunk 1 above)
Comment (robert-zaremba) [reply packages/algob/src/lib/dryrun.ts:188]: OK, let's add a comment to explain that

**Review by robert-zaremba — APPROVED** [review]

**Comment by robert-zaremba** [comment]
Hmm, ok the changes bare there, but I don't know why the test fails. We use pipenv run, so it should work.

File: packages/algob/src/lib/dryrun.ts (hunk 1 above)
Comment (ratik21) [reply packages/algob/src/lib/dryrun.ts:188]: ok

**Pull request merged**